File: LC/python/Sum Root to Leaf Numbers.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def sumNumbers(self, root):
        def dfs(root,Sum):
            if root == None:
                return
            Sum = Sum*10+root.val
            if root.left == None and root.right == None:
                self.res += Sum
            else:
                dfs(root.left,Sum)
                dfs(root.right,Sum)
        self.res = 0
        dfs(root,0)
        return self.res
        

File: LC/java/Sum Root to Leaf Numbers.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public int sumNumbers(TreeNode root) {
        if(root==null){
            return 0;
        }
        
        return dfs(root,0);
    }
    
    public int dfs(TreeNode n,int sum){
        if(n == null){
            return 0;
        }
        sum = sum*10+n.val;
        
        if(n.left==null && n.right==null){
            return sum;
        }
        
        return dfs(n.left,sum)+dfs(n.right,sum);
    }
}